pgrx/src/datum/mod.rs

```diff
@@ -149,3 +149,20 @@ impl<'src> Datum<'src> {
 /// A tagging trait to indicate a user type is also meant to be used by Postgres
 /// Implemented automatically by `#[derive(PostgresType)]`
 pub trait PostgresType {}
+
+/// Creates a [`Vec<pg_sys::Oid>`] containing identifiers of the provded types.
+///
+/// # Examples
+///
+/// ```
+/// use pgrx::{oids_of, datum::IntoDatum};
+/// let v = oids_of![i32, f64];
+/// assert_eq!(v[0], i32::type_oid().into());
+/// assert_eq!(v[1], f64::type_oid().into());
+/// ```
+#[macro_export]
+macro_rules! oids_of {
+    ($($t:path),* $(,)?) => ({
+        vec![$(::pgrx::pg_sys::PgOid::from(<$t>::type_oid())),*]
```

Comment:
I don't think this should include a `vec![]` when it could instead be invoked as `vec![oids_of!(...)]`

this would allow constructing arrays also, or other structures.

Reply:
Yes, that's the plan when I get rid of `Option<Vec<PgOid>>` because now it's too wordy.

Reply:
...if that's the plan, then?

Reply:
Yeah, I will remove `vec!` from here and will change the methods of `SpiClient` to use slices as soon as you that the breaking change time has come (:

Reply:
I'd rather not add this then just break it again.

Reply:
Let it be non-breaking then.